pkg/services/query/query.go

```diff
@@ -304,7 +342,7 @@ func (s *Service) parseMetricRequest(ctx context.Context, user *user.SignedInUse
 		req.httpRequest = reqDTO.HTTPRequest
 	}
 
-	return req, nil
+	return req, req.validateRequest()
```

Comment:
I'm not sure, but think this could break the public dashboards feature, i.e. if the public dashboard feature not are using the `DataSourceWithBackend` class to make queries. Ping @grafana/grafana-partnerships-team

Reply:
As implemented the validation only happens when the headers exist -- the only issue we will have is if they are getting sent and used for something different.

As implemented, this does not *require* anyone to set the headers.  Lets get routing/auditing working before the next pass of making sure everywhere uses it consistently

Reply:
Nice, didn't notice :+1: